### ISSUE TITLE
Drop own module map or include framework keyword in module maps

### ISSUE THREAD

**Issue by jvigneshcs** [Azure/azure-iot-sdk-c#2064]
**Is your feature request related to a problem? Please describe.**
I have posted my problem [here](https://github.com/CocoaPods/CocoaPods/issues/10547) and [here](https://github.com/Azure-Samples/azure-iot-samples-ios/issues/18)

**Describe the solution you'd like**
Basically mine is not a solution that I am suggesting, it is more like to know the technical possibility from your end whether this suggestion would be implemented or is it something not even remotely possible.

Based on the suggestion mentioned [here](https://github.com/CocoaPods/CocoaPods/issues/10547#issuecomment-906553724),
Is it possible to drop own module map or update it to include the `framework` keyword in your module maps?

**Describe alternatives you've considered**

**Additional context**
More informations provided in the above referenced issues.

**Comment by danewalton** [comment]
Hi @jvigneshcs is this essentially the change you are looking for in the podfiles?

From this:

![Screen Shot 2021-09-16 at 5 31 39 PM](https://user-images.githubusercontent.com/36710865/133705662-aaee70bc-c2f9-4a32-bcec-4506d7bda70c.png)

To this:

![Screen Shot 2021-09-16 at 5 32 09 PM](https://user-images.githubusercontent.com/36710865/133705661-d7af596f-0c0d-4d69-a986-1449de2ec0ad.png)

That's what I'm seeing for example here: https://stackoverflow.com/questions/34478180/modulemap-path-resolution

**Comment by jvigneshcs** [comment]
Hi @danewalton yes you are correct.

**Comment by danewalton** [comment]
Okay pods have been updated:

Macro Utils: [1.3.9-patch-2](https://github.com/CocoaPods/Specs/commit/2451b468fe81c8680dae26646aafcc196af8d839)
UMock: [1.3.9-patch-4](https://github.com/CocoaPods/Specs/commit/652bbece81f0338032ef7098d240cc0af7564536)
C-Utility: [1.8.0](https://github.com/CocoaPods/Specs/commit/02734ff21bd96d42a2447295ee3e1cbfe4742ce8)
uMQTT: [1.8.0](https://github.com/CocoaPods/Specs/commit/a8ffb364c9e5aa7797921aaaca2fd6c3886f3f01)
uAMQP: [1.8.0](https://github.com/CocoaPods/Specs/commit/ea08ad7c6943a7c57cf35c2352b2ce24bbd6c20d)
C SDK: [1.8.0](https://github.com/CocoaPods/Specs/commit/9d4539b58f91ad21e3415cb9660da6902903109e)

I haven't done full validation yet but please feel free to try these and see if it works for you.

**Comment by danewalton** [comment]
I'm going to consider this closed since the original ask looks to be taken care of. Please let us know if you would like it reopened.

**Comment by danewalton** [comment]
This had to be reverted due to #2109 

Based on all the research I could do I found this to summarize:
https://newbedev.com/swift-compiler-error-non-modular-header-inside-framework-module

I tried resolving the build errors a number of ways and tried the route of just removing the module map and letting Cocoapods autogenerate one. None of it worked.

For now we won't support it unless a suggested resolution can be found.
